### PR TITLE
[Remote Snapshot Sharing] Add metric for ratio of dirty chunks in a copy-on-write snapshot

### PR DESCRIPTION
**Pull request by maggie-lou**
**Related issues**: N/A

### CODE REVIEW

**Comment by maggie-lou** [comment]
Current dependencies on/for this PR:
* master
  * **PR #4692** <a href="https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/4692" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/4692?utm_source=stack-comment).

File: server/metrics/metrics.go [hunk 1]
```diff
@@ -1713,6 +1737,20 @@ var (
 		CompressionType,
 	})
 
+	// #### Examples
+	//
+	// ```promql
+	// # Histogram buckets with the count of elements in each compression ratio bucket
+	// # Visualize with the Bar Gauge type
+	// # Legend: {{le}}
+	// # Format: Heatmap
```
Comment (bduffany) [inline server/metrics/metrics.go:1746]: ditto

**Review by bduffany — APPROVED** [review]

File: server/metrics/metrics.go [hunk 2]
```diff
@@ -959,6 +962,27 @@ var (
...
+	COWSnapshotDirtyChunkRatio = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: bbNamespace,
+		Subsystem: "firecracker",
+		Name:      "cow_snapshot_dirty_chunk_ratio",
+		Buckets:   prometheus.LinearBuckets(0, .05, 20),
+		Help:      "After a copy-on-write snapshot has been used, the ratio of dirty/total chunks.",
+	}, []string{
+		GroupID,
+		FileName,
+	})
```
[the remark this answers is not in the document]
Comment (maggie-lou) [reply server/metrics/metrics.go:974]: Done

File: server/metrics/metrics.go [hunk 3]
```diff
@@ -959,6 +962,27 @@ var (
...
+
+	// #### Examples
+	//
+	// ```promql
+	// # To view how many elements fall into each bucket
+	// # Visualize with the Bar Gauge type
+	// # Legend: {{le}}
+	// # Format: Heatmap
```
[the remark this answers is not in the document]
Comment (maggie-lou) [reply server/metrics/metrics.go:982]: I find the prometheus histograms really hard to use. Typically when I'm looking into a metric and struggling with how to write a query, I'll refer to metrics.go to see if there are any example queries to get me started. Personally I find this useful, but if you think it'll create extra support burden if customers start to rely on these examples I can remove this

**Review by bduffany — APPROVED** [review]

**Pull request merged**